pkg/metrics/observers/skipper.go

```diff
@@ -0,0 +1,110 @@
+package observers
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	flaggerv1 "github.com/weaveworks/flagger/pkg/apis/flagger/v1beta1"
+	"github.com/weaveworks/flagger/pkg/metrics/providers"
+)
+
+var skipperQueries = map[string]string{
+	"request-success-rate": `
+	{{- $route := printf "kube_%s__%s.*__%s" namespace ingress service }}
```

Comment:
This is trying to match the route ID, right?
We always thought of it as an internal detail.

@aryzska, please see this. There is now an external dependency and if we break this we should increase skipper non patch version to highlight this.

Reply:
@szuecs That is the only way we found to filter Prometheus metrics correctly. 

As an idea would it make sense to split this string into distinctive objects that are less error prone!?

Reply:
@universam1 thinking about it, it would make sense to have also `kube(ew)?_%s__%s.*__%s(_[0-9]+)?`
Like this the east west routes `(ew)?` and the skipper-routes annotation `(_[0-9]+)?` should be hit, too.

Reply:
Good feedback! :heart_decoration: 
thanks - implemented https://github.com/o11n/flagger/pull/5